src/Core/Module/Legacy/ModuleInterface.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Module\Legacy;
+
+/**
+ * Define what should be a module.
+ * Note:We don't typeHint on old Module class to not create hard dependency with Legacy.
+ */
+interface ModuleInterface
+{
+}
```

Comment:
Such interface would be more useful with a contract definition.
Could be an excellent opportunity to provide getter methods to replace direct access to public properties:
```suggestion
interface ModuleInterface
{
    // Access to legacy instance
    public function getInstance(string $serviceId): Module;

    // Dedicated method to fetch attributes instead of direct access to public property "attributes"
    public function getAttributes(): ParameterBag;
}
```

Reply:
There's actually 2 `ModuleInterface`, this one is in the `Legacy` namespace and should not contains method definitions for backward compatibility reasons. Also, I'm not sure those attributes should be part of the interface, it's more related to the implementation IMO 🤔